packages/color-picker/src/components/alpha-slider.vue

```diff
@@ -0,0 +1,144 @@
+<template>
+  <div class="el-color-alpha-slider" :class="{ 'is-vertical': vertical }">
+    <div
+      ref="bar"
+      class="el-color-alpha-slider__bar"
+      :style="{
+        background: background
+      }"
+      @click="handleClick"
+    >
+    </div>
+    <div
+      ref="thumb"
+      class="el-color-alpha-slider__thumb"
+      :style="{
+        left: thumbLeft + 'px',
+        top: thumbTop + 'px'
+      }"
+    >
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import { defineComponent, watch, ref, onMounted, getCurrentInstance, PropType } from 'vue'
+import type Color from '../color'
+import draggable from '../draggable'
+
+export default defineComponent({
+  name: 'ElColorAlphaSlider',
+  props: {
+    color: {
+      type: Object as PropType<Color>,
+      required: true,
+    },
+    vertical: {
+      type: Boolean,
+      default: false,
+    },
+  },
+  setup(props) {
+    const instance = getCurrentInstance()
+    // ref
+    const thumb = ref<Nullable<HTMLElement>>(null)
+    const bar = ref<Nullable<HTMLElement>>(null)
+
+    // data
+    const thumbLeft = ref(0)
+    const thumbTop = ref(0)
+    const background = ref<Nullable<string>>(null)
+
+    watch(() => props.color.get('alpha'), () => {
+      update()
+    })
+    watch(() => props.color.value, () => {
+      update()
+    })
+
+    //methods
+    function getThumbLeft() {
+      if (props.vertical) return 0
+      const el = instance.vnode.el
+      const alpha = props.color.get('alpha')
+
+      if (!el) return 0
+      return Math.round(alpha * (el.offsetWidth - thumb.value.offsetWidth / 2) / 100)
+    }
+
+    function getThumbTop() {
+      if (!props.vertical) return 0
+      const alpha = color.get('_alpha')
```

Comment:
is here correctly ?
i saw the `color.ts` code
```
  get(prop: string) {
    return this['_' + prop]
  }
```
this line will become `color.__alpha` ?
emmm... where is `color` define?  maybe `props.color`?